contribs/emissions/src/main/java/org/matsim/contrib/emissions/ColdEmissionHandler.java

```diff
@@ -86,19 +86,14 @@ public void reset(int iteration) {
 
     @Override
     public void handleEvent(LinkLeaveEvent event) {
+
+        // extract event details
 		Id<Vehicle> vehicleId = event.getVehicleId();
         Id<Link> linkId = event.getLinkId();
         Link link = this.scenario.getNetwork().getLinks().get(linkId);
         double linkLength = link.getLength();
 
-        if (linkLength == 0.) {
-            if (zeroLinkLengthWarnCnt == 0 ){
-                logger.warn("Length of the link "+ linkId + " is zero. No emissions will be estimated for this link. Make sure, this is intentional.");
-                logger.warn(Gbl.ONLYONCE);
-                zeroLinkLengthWarnCnt++;
-            }
-            return;
-        }
+        warnIfZeroLinkLength(linkId, linkLength);
 
         Double previousDistance = this.vehicleId2accumulatedDistance.get(vehicleId);
```

Comment:
this should be lower case `double`

Reply:
I'd rather not change this, because if changed to `double`, it cannot be compared to `null` anymore and it seems like the `!= 0` test is a good "safety check". It is also used in the `coldEmissions` map within the `if` statement... so it doesn't seem like it was a typo by the original author?